src/pyhf/cli/contrib.py

```diff
@@ -0,0 +1,62 @@
+"""CLI for functionality that will get migrated out eventually."""
+import logging
+
+import click
+import urllib.request
+import tarfile
+from io import BytesIO
+from pathlib import Path
+
+from .. import exceptions
+
+logging.basicConfig()
+log = logging.getLogger(__name__)
+
+
+@click.group(name="contrib")
+def cli():
+    """Contrib experimental operations"""
+
+
+@cli.command()
+@click.argument("archive-url", default="-")
+@click.argument("output-directory", default="-")
+@click.option("-v", "--verbose", is_flag=True, help="Enables verbose mode")
+@click.option(
+    "-f", "--force", is_flag=True, help="Force download from non-approved host"
+)
+def download(archive_url, output_directory, verbose, force):
+    """
+    Download the patchset archive from the remote URL and extract it in a directory at the path given.
+
+    Example:
+
+    .. code-block:: shell
+
+        $ pyhf contrib download --verbose https://www.hepdata.net/record/resource/1408476?view=true 3L-likelihood
+
+        \b
+        3L-likelihoods/patchset.json
+        3L-likelihoods/README.md
+        3L-likelihoods/BkgOnly.json
+
+    Raises:
+        :class:`~pyhf.exceptions.InvalidArchiveHost`: if the provided archive host name is not known to be valid
+    """
+    if not force:
+        hostname = archive_url.split("://")[1].split("/")[0].strip("www.")
+        valid_hosts = ["hepdata.net", "doi.org"]
+        if hostname not in valid_hosts:
+            raise exceptions.InvalidArchiveHost(
+                f"{hostname} is not an approved archive host: {', '.join(str(host) for host in valid_hosts)}\n"
+                + "To download an archive from this host use the --force option."
+            )
+
+    req = urllib.request.Request(archive_url)
+    with urllib.request.urlopen(req) as response:
```

Comment:
this one might necessitate a `requests` dependency. I'm not so sure what we want to do here, as this is a medium severity security issue anyway...

Reply:
Yeah, I'm not sure either. While I'm not against adding it, I'm hesitant to add things as core dependencies of `pyhf` especially for something in the contrib area. Should I just add it to the `contrib` extra?

Reply:
Er, no this breaks the CLI now. This would require adding requests as a real dependency. Thoughts?

Reply:
We can do this in contrib as a lazy dependency. Move the `import requests` to become a

```
try:
  import requests
  # add in command
except:
  pass
```